Sources/SwiftLexicalLookup/SwiftLexicalLookup.docc/LookupRules.md

```diff
@@ -0,0 +1,78 @@
+
+## Overview
+Unqualified lookup can be initiated from any source position, optionally targeting a specific identifier for name comparison. Unqualified lookup traverses the syntax tree from bottom-up, collecting names that match the lookup criteria. A name matches the lookup if the specified identifier (if present) refers to the introduced name and the position from which the unqualified lookup was triggered is preceded by the position where the name was introduced (if specified).
+
+## Default Scope Behavior
+Names are introduced and their availabilities are set by special syntax nodes referred to as syntax scopes or simply scopes. By default, each scope for lookup produces a result object that associates the scope of origin with the matched names. Lookup is then invoked on the parent scope, i.e., the closest syntax scope ancestor of the original scope in the syntax tree.
+
+## Name Kinds
+There are three distinct kinds of names identified by unqualified lookup:
+- **Identifier**: Represents `IdentifiableSyntax` syntax nodes introducing names such as `IdentifierPatternSyntax`, `ClosureParameterSyntax`, or `FunctionParameterSyntax`.
+- **Declaration**: Represents `NamedDeclSyntax` syntax nodes introducing names such as `ClassDeclSyntax`, `FunctionDeclSyntax`, or `ProtocolDeclSyntax`.
+- **Implicit**: Represents names introduced by some scopes implicitly, without an associated token representation. Such names include `self`, `newValue`, or `error`.
+
+## Specialized Scopes
+Some scopes share common behavior and can be further generalized as follows:
+- **Type Scope** (`TypeScopeSyntax`): Implicitly introduces `self` and `Self` names.
+- **Sequential Scope** (`SequentialScopeSyntax`): Interleaves its own results with results produced by introducing to sequential parent scopes when encountered during name extraction. Results are ordered from closest to furthest from the unqualified lookup position.
+- **Introducing to Sequential Parent Scope** (`IntroducingToSequentialParentScopeSyntax`): Produces a result that is later interleaved with the results produced by its sequential parent.
+
+> Example of Sequential Scope behavior (code block):
+> ```swift
+> func foo(x: Int?) {
+>   // Guard scope
+>   guard let a = x else { return }
+> 
+>   // In code block scope
+>   let a: Int? = a
+> 
+>   // Guard scope
+>   guard let b = x, let a else { return }
+> 
+>   // In code block scope
+>   let a = b
+> 
+>   a // <-- lookup here
+>   b // <-- lookup here
+> }
+> ```
+> When looking up the reference `a`, we get 4 results in total, going from the bottom to the top of the source code: the `a` declaration associated with the function's code block body scope, the `a` variable associated with the bottom-most `guard`, another `a` declaration associated with the code block, and finally the `a` introduced in the top-most `guard`.
+> When looking up `b`, on the other hand, we get just one result — the `b` declaration inside the bottom-most `guard`.
```

Comment:
```suggestion
> When looking up `b`, on the other hand, we get just one result: the `b` declaration inside the bottom-most `guard`.
```
```suggestion
> When looking up `b`, on the other hand, we get just one result — the `b` declaration inside the bottom-most `guard`.
```


I would use `–` or `:` consistently between the this sentence and the one above.